units/systemd-resolv-conf.service

```diff
@@ -0,0 +1,27 @@
+#  SPDX-License-Identifier: LGPL-2.1-or-later
+#
+#  This file is part of systemd.
+#
+#  systemd is free software; you can redistribute it and/or modify it
+#  under the terms of the GNU Lesser General Public License as published by
+#  the Free Software Foundation; either version 2.1 of the License, or
+#  (at your option) any later version.
+
+[Unit]
+Description=Create resolv.conf if not already exists
+Documentation=man:systemd-resolved.service(8) man:resolv.conf(5)
+After=systemd-journald.service
+After=local-fs.target network-pre.target systemd-sysusers.service
+; no dynamic DNS configuration would be known before network configuration
+After=systemd-networkd.service
```

Comment:
All this ordering is unecessary: the service does not *use* any information about network configuration (or the network). It is enough that it is ordered after `systemd-resolve.service`.

Reply:
But it were ment to be run always Before systemd-resolve.service, but AFTER systemd-networkd.service (and similar). It is meant to give a chance to DHCP of network connection to prepare resolv.conf instead. This should work as fallback, not the first instance.

Reply:
I admit it might run also After=systemd-resolved.service, if resolved does not use it for anything. It does detect it on startup, but I am not sure what action depends on that detection result. It has to run Before=nss-lookup.target. Adding that to resolv.conf creation service.